guides/source/getting_started.md

```diff
@@ -1991,6 +1991,191 @@ the `app/views/comments` directory.
+
+  def archived? 
+    status == 'archived'
+  end
+end
+```
+
+and in the `Comment` model:
+
+```ruby
+class Comment < ApplicationRecord
+  belongs_to :article
+
+  VALID_STATUSES = ['public', 'private', 'archived']
+
+  validates :status, in: VALID_STATUSES
+
+  def archived?
+    status == 'archived'
+  end
+end
+```
+
+Then, in our `index` action template (`app/views/articles/index.html.erb`) we would use the `archived?` method to avoid displaying any article that is archived:
+
+```html+erb
+<h1>Listing Articles</h1>
+<%= link_to 'New article', new_article_path %>
+<table>
+  <tr>
+    <th>Title</th>
+    <th>Text</th>
+    <th colspan="3"></th>
+  </tr>
+
+  <% @articles.each do |article| %>
+    <% unless article.archived? %> 
+      <tr>
+        <td><%= article.title %></td>
+        <td><%= article.text %></td>
+        <td><%= link_to 'Show', article_path(article) %></td>
+        <td><%= link_to 'Edit', edit_article_path(article) %></td>
+        <td><%= link_to 'Destroy', article_path(article),
+                method: :delete,
+                data: { confirm: 'Are you sure?' } %></td>
+      </tr>
+    <% end %>
+  <% end %>
+</table>
+```
+
+However, if you look again at our models now, you can see that the logic is duplicated. If in future we increase the functionality of our blog - to include private messages, for instance -  we might find ourselves duplicating the logic yet again. This is where concerns come in handy.
+
+Let's call our new concern (module) `Visible`, a module to use for any model that has a status. We can create a new file inside `app/models/concerns` called `statuslike.rb` , and store all of the status methods that were duplicated in the models.
+
+`app/models/concerns/statuslike.rb`
+
+```ruby
+module Visible
+  def archived?
+    status == 'archived'
+  end
+end
+```
+
+We can add our status validation to the concern, but this is slightly more complex as validations are methods called at the class level. The `ActiveSupport::Concern` ([API Guide](https://api.rubyonrails.org/classes/ActiveSupport/Concern.html)) gives us a simpler way to include them:
+
+```ruby
+module Visible
+  extends ActiveSupport::Concern
+
+  included do
+    VALID_STATUSES = ['public', 'private', 'archived']
+
+    validates :status, in: VALID_STATUSES
+  end
+
+  def archived?
+    status == 'archived'
+  end
+end
+```
+
+Now, we can remove the duplicated logic from each model and instead include our new `Visible` module:
+
+
+In `app/models/article.rb`:
+```ruby
+class Article < ApplicationRecord
+  include Visible
+  has_many :comments
+
+  validates :title, presence: true,
+                    length: { minimum: 5 }
+
+end
+```
+
+and in `app/models/comment.rb`:
+
+```ruby
+class Comment < ApplicationRecord
+  include Visible
+  belongs_to :article
+end
+```
+
+Class methods can also added to concerns. If we want a count of public articles or comments to display on our main page, we might add a class method to Visible as follows:
```

Comment:
@dzunk I added this section regarding class methods - I admit to struggling with an example that works in the context of this Getting Started blog. I would have preferred to make a class method that retrieves all public articles/comments by user, but there is no `user` object in this blog app - the closest we get to that is having a `commenter`.  Would be great to hear your thoughts on this!

Reply:
I think it's a fine example, although you might want to update the logic to actually `.count` the records 😉 

An alternative could be something related to the state transitions:
```ruby
class_methods do
  def archive_all
    update_all(status: 'archived')
  end
end
```

Reply:
Ahh, good catch! 😅  I quite like your alternative method, but it seems to me like the usage example would be a little more complicated (one would probably use the method in a new action) so I'll keep the `count` example (and make sure it actually does count... 🙈 )